src/components/richText/LexicalRichText.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  LexicalComposer,
+  InitialConfigType,
+} from "@lexical/react/LexicalComposer";
+import { RichTextPlugin } from "@lexical/react/LexicalRichTextPlugin";
+import { ContentEditable } from "@lexical/react/LexicalContentEditable";
+import LexicalErrorBoundary from "@lexical/react/LexicalErrorBoundary";
+import { TableCellNode, TableNode, TableRowNode } from "@lexical/table";
+import { ListItemNode, ListNode } from "@lexical/list";
+import { CodeHighlightNode, CodeNode } from "@lexical/code";
+import { AutoLinkNode, LinkNode } from "@lexical/link";
+import { HeadingNode, QuoteNode } from "@lexical/rich-text";
+import { HashtagNode } from "@lexical/hashtag";
+import { EditorThemeClasses } from "lexical";
+import DefaultNodeStyling from "./DefaultStyling";
+import { ImageNode } from "./imageNode";
+
+/**
+ * The shape of data passed to {@link LexicalRichText}.
+ */
+export interface LexicalRichTextProps {
+  serializedAST: string;
+  nodeClassNames?: EditorThemeClasses;
+}
+
+/**
+ * Renders a read-only view of a Lexical Rich Text field. Styling for the various
+ * types of Rich Text element can be optionally provided. If not provided, Yext default
+ * styling will be applied.
+ */
+export function LexicalRichText({
+  serializedAST,
+  nodeClassNames,
+}: LexicalRichTextProps) {
+  return (
+    <LexicalComposer
+      initialConfig={generateConfig(serializedAST, nodeClassNames)}
+    >
+      <RichTextPlugin
+        contentEditable={<ContentEditable className="editor-input" />}
```

Comment:
what does this prop do? it sounds like it would make the rich text content editable, which we don't want

Reply:
This is a case of bad naming on Lexical's part. The core concept in Lexical is the Editor, which is headless. The Editor is configured to allow/disallow modification of its State (the content). This is the only way to set "read-only". In the `generateConfig` function, I'm setting the Editor to have `editable` as `false`. The `contentEditable` prop specifies what Component/DOM will be used to display the Editor's state.